lib/api/bucketPut.js

```diff
@@ -174,60 +218,37 @@ function bucketPut(authInfo, request, log, callback) {
             if (!authInfo.isRequesterAnIAMUser()) {
                 return next(null, locationConstraint);
             }
-
             const authParams = auth.server.extractParams(request, log, 's3', request.query);
-            const ip = requestUtils.getClientIp(request, config);
-            const baseParams = {
-                authParams,
-                ip,
-                bucketName,
-                request,
-                authInfo,
-                locationConstraint,
-            };
-            const requestConstantParams = [Object.assign(
-                baseParams,
-                { apiMethod: 'bucketPut' },
-            )];
-
-            if (!_isObjectLockEnabled(request.headers)) {
-                requestConstantParams.push(Object.assign(
-                    baseParams,
-                    { apiMethod: 'bucketPutObjectLock' },
-                ));
-                requestConstantParams.push(Object.assign(
-                    baseParams,
-                    { apiMethod: 'bucketPutVersioning' },
-                ));
```

Comment:
I think this will fix it.
```js
requestConstantParams.push(Object.assign(
    {},
    baseParams,
    { apiMethod: 'bucketPutObjectLock' },
));
```